src/org/jgroups/util/DefaultSocketFactory.java

```diff
@@ -1,113 +1,94 @@
+        return new Socket(address, port, localAddr, localPort);
     }
 
     public ServerSocket createServerSocket(String service_name) throws IOException {
-        return add(new ServerSocket(), service_name);
+        return new ServerSocket();
     }
 
     public ServerSocket createServerSocket(String service_name, int port) throws IOException {
-        return add(new ServerSocket(port), service_name);
+        return new ServerSocket(port);
     }
 
     public ServerSocket createServerSocket(String service_name, int port, int backlog) throws IOException {
-        return add(new ServerSocket(port, backlog), service_name);
+        return new ServerSocket(port, backlog);
     }
 
     public ServerSocket createServerSocket(String service_name, int port, int backlog, InetAddress bindAddr) throws IOException {
-        return add(new ServerSocket(port, backlog, bindAddr), service_name);
+        return new ServerSocket(port, backlog, bindAddr);
     }
 
     public DatagramSocket createDatagramSocket(String service_name) throws SocketException {
-        return add(new DatagramSocket(), service_name);
+        return new DatagramSocket();
     }
 
     public DatagramSocket createDatagramSocket(String service_name, SocketAddress bindaddr) throws SocketException {
-        return add(new DatagramSocket(bindaddr), service_name);
+        return new DatagramSocket(bindaddr);
     }
 
     public DatagramSocket createDatagramSocket(String service_name, int port) throws SocketException {
-        return add(new DatagramSocket(port), service_name);
+        return new DatagramSocket(port);
     }
 
     public DatagramSocket createDatagramSocket(String service_name, int port, InetAddress laddr) throws SocketException {
-        return add(new DatagramSocket(port, laddr), service_name);
+        return new DatagramSocket(port, laddr);
     }
 
     public MulticastSocket createMulticastSocket(String service_name) throws IOException {
-        return add(new MulticastSocket(), service_name);
+        return new MulticastSocket();
     }
 
     public MulticastSocket createMulticastSocket(String service_name, int port) throws IOException {
-        return add(new MulticastSocket(port), service_name);
+        return new MulticastSocket(port);
     }
 
     public MulticastSocket createMulticastSocket(String service_name, SocketAddress bindaddr) throws IOException {
-        return add(new MulticastSocket(bindaddr), service_name);
+        return new MulticastSocket(bindaddr);
     }
 
     public void close(Socket sock) throws IOException {
-        remove(sock);
         Util.close(sock);
     }
 
     public void close(ServerSocket sock) throws IOException {
-        remove(sock);
         Util.close(sock);
     }
 
     public void close(DatagramSocket sock) {
-        remove(sock);
         Util.close(sock);
     }
 
-    public Map<Object, String> getSockets() {
-        return sockets;
+    public Map<Object,String> getSockets() {
```

Comment:
This method is called only by JChannel.MyProbeHandler.getOpenSockets() which is removed. 
I would remove it from DefaultSocketFactory and SocketFactory.